modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/KotlinClientCodegen.java

```diff
@@ -264,6 +291,26 @@ public void processOpts() {
             LOGGER.warn(CodegenConstants.INVOKER_PACKAGE + " with " + this.getName() + " generator is ignored. Use " + CodegenConstants.PACKAGE_NAME + ".");
         }
 
+        if (additionalProperties.containsKey(DATE_LIBRARY)) {
+            setDateLibrary(additionalProperties.get(DATE_LIBRARY).toString());
+        }
+
+        if (DateLibrary.THREETENBP.value.equals(dateLibrary)) {
+            additionalProperties.put(DateLibrary.THREETENBP.value, "true");
```

Comment:
additionalProperties should be true here rather than "true" if it's intended to be used for switching in templates.